src/convertToObject.js

```diff
@@ -6,7 +6,12 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  return sourceString
+    .split(';')
+    .filter((item) => item.includes(':'))
+    .map((item) => item.split(':'))
+    .map(([property, value]) => [property.trim(), value.trim()])
+    .reduce((acc, [property, value]) => ({ ...acc, [property]: value }), {});
```

Comment:
Using the spread operator inside the reduce callback is not recommended for performance reasons, as it copies the accumulator object on each iteration. Instead, you can assign the new property directly to the accumulator object.